### ISSUE TITLE
podman pull fails with "permission denied" in rootless mode with overlay

### ISSUE THREAD

**Issue by dbuse** [containers/libpod#1949]
**Is this a BUG REPORT or FEATURE REQUEST?:**

[x]: # **kind bug**

**Description**

When using podman in rootless mode with overlayfs as the storage driver, pulling some images fails with a "permission denied" error.
Note: It does not matter if the storage location (graphroot) is non-existent oder pre-populates with working images.

**Steps to reproduce the issue:**

1. podman pull opensuse/portus

**Describe the results you received:**
```
Trying to pull docker.io/opensuse/portus:latest...Getting image source signatures
Copying blob sha256:cc56dde21ac8523812d9849463cca9cff6ed4412d51be0ca7faa3fd392265a14
 39.29 MB / 39.29 MB [======================================================] 2s
Copying blob sha256:a072addd01aeffc866ff7c0fce9d7145d15c0cd126d4d076a2ab608ae58cff71
 1.46 KB / 1.46 KB [========================================================] 0s
Copying blob sha256:ac98668b5211ab8aa28a4e1e164eb8848639484654c55abcd6ad62f8b4c146ca
 55.99 MB / 55.99 MB [======================================================] 2s
Copying config sha256:dc91deeead11ac2ad3f5f476b0446ff649c461eb46aacfcc24bf2ad816ce0b94
 3.95 KB / 3.95 KB [========================================================] 0s
Writing manifest to image destination
Storing signatures
Failed
error pulling image "opensuse/portus": unable to pull opensuse/portus: 1 error occurred:

* Error committing the finished image: error adding layer with blob "sha256:ac98668b5211ab8aa28a4e1e164eb8848639484654c55abcd6ad62f8b4c146ca": Error processing tar file(exit status 1): operation not permitted
```

**Describe the results you expected:**
Should download and unpack the image/layers to be ready for further use.

**Additional information you deem important (e.g. issue happens only occasionally):**
1.) When running as root or with `vfs` as storage driver (in a different graphroot), pulling and running the image works as expected.
2.) Other images may work, e.g., `busybox` or `ubuntu:16.04`:
```
podman pull busybox 
Trying to pull docker.io/busybox:latest...Getting image source signatures
Copying blob sha256:90e01955edcd85dac7985b72a8374545eac617ccdddcc992b732e43cd42534af
 710.92 KB / 710.92 KB [====================================================] 0s
Copying config sha256:59788edf1f3e78cd0ebe6ce1446e9d10788225db3dedcfd1a59f764bad2b2690
 1.46 KB / 1.46 KB [========================================================] 0s
Writing manifest to image destination
Storing signatures
59788edf1f3e78cd0ebe6ce1446e9d10788225db3dedcfd1a59f764bad2b2690
```

**Output of `podman version`:**

```
Version:       0.11.2-dev
Go Version:    go1.10.4
OS/Arch:       linux/amd64
```

**Output of `podman info`:**

```
host:
  BuildahVersion: 1.5-dev
  Conmon:
    package: 'cri-o-1.12: /usr/lib/crio/bin/conmon'
    path: /usr/lib/crio/bin/conmon
    version: 'conmon version 1.12.4-dev, commit: '
  Distribution:
    distribution: ubuntu
    version: "16.04"
  MemFree: 12123840512
  MemTotal: 16499482624
  OCIRuntime:
    package: 'cri-o-runc: /usr/lib/cri-o-runc/sbin/runc'
    path: /usr/lib/cri-o-runc/sbin/runc
    version: 'runc version spec: 1.0.1-dev'
  SwapFree: 0
  SwapTotal: 0
  arch: amd64
  cpus: 8
  hostname: heinz
  kernel: 4.4.0-139-generic
  os: linux
  rootless: true
  uptime: 20h 29m 41.33s (Approximately 0.83 days)
insecure registries:
  registries: null
registries:
  registries:
  - docker.io
store:
  ContainerStore:
    number: 0
  GraphDriverName: overlay
  GraphOptions: []
  GraphRoot: /scratch/containers/buse
  GraphStatus:
    Backing Filesystem: extfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
  ImageStore:
    number: 1
  RunRoot: /run/user/10078
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
Running on a physical machine, Ubuntu 16.04, libpod, buildah etc. installed from projectatomic ppa.
$HOME is on a NFS share, so I moved graphroot to /scratch

**Comment by rhatdan** [comment]
Do you have newuidmap installed?
@giuseppe @mheon Is this the same issue others are having with Ubuntu?

**Comment by giuseppe** [comment]
this was recently fixed upstream with: https://github.com/containers/storage/pull/237

we have already vendored that change in podman

**Comment by dbuse** [comment]
> Do you have newuidmap installed?

Yes, installed and configuration set in `/etc/subuid` and `/etc/subgid`.

**Comment by baude** [comment]
i thought one couldn't use overlay and rootless?

**Comment by rhatdan** [comment]
Fixed in podman 0.12.1
